### PR TITLE
Preliminary APK Release for Kenya eCHIS

### PR DESCRIPTION
**Pull request by kennsippell**
This is an effort to quickly release a single APK for eCHIS Kenya with minimal code change. This PR is not intended to merge. 
This should have the same functional result as the PR which is implemented generically and is intended to merge #306 .
Device provisioning begins May 18th and it is valuable to have users on a single APK from the beginning.

### CODE REVIEW

**Comment by kennsippell** [comment]
This is the minimum code-change possible. SharedPreferences is not an ordered datastructure, so one addition we should consider is to sort the rows by name before displaying them.  image

**Comment by jkuester** [comment]
[echis_kenya.zip](https://github.com/medic/cht-android/files/11502595/echis_kenya.zip)

Attached a zip of the APKs that I built locally from the latest commit out here.

**Comment by mrjones-plip** [comment]
[This build](https://github.com/medic/cht-android/files/11502930/echis_kenya.zip) fails to register the right URL intents in the android app such that Magic Links don't work.  Here's how I tested:
 
 1. start up a 4.1.x instance with Docker Helper
 2. using the default app's `[app_settings.json](https://github.com/medic/cht-core/blob/master/config/default/app_settings.json)`, edit that to enable SMS Login:
      ```json
      "token_login": {
        "enabled": true,
        "translation_key": "sms.token.login.help"
      }
      ```
3. use `cht-conf` to push app settings to instance: `cht --url=https://medic:password@192-168-68-26.local-ip.medicmobile.org upload-app-settings`
4. As `medic` admin user, Create a new offline user and specify "use SMS login"
5. As `medic` admin user, go to "outbound messages", get the URL of with the magic link
6. create a web page with the link on my desktop (`vim index.html` followed by `python3 -m http.server 8000`)
7. browse to my desktop's `IP:8000` on my phone
8. Click the link

Expected: APK catches the URL intent and CHT Android loads
Actual: URL loads in browser

ANdroid 5.1, Moto X, WebView 95

Just in case it matters, I'll try sending an SMS to the device using Google Voice

**Comment by mrjones-plip** [comment]
@kennsippell  - I noticed the order of the counties changes when selecting one and going back (5 two finger taps, followed by two finger right swipe - faster than an app reset!)

You have a todo of "sort the county list by name", so calling this out:

https://github.com/medic/cht-android/assets/8253488/8efc4354-f911-4610-a3b9-4f12bf71e1cc

**Comment by jkuester** [comment]
FTR, I did confirm that this error (that we were discussing @mrjones-plip) is also displayed on master when one of the URLs is invalid.  Not great behavior, but it is existing behavior and not an issue with the code changes.  Not something that needs to be addressed here.

image

**Comment by mrjones-plip** [comment]
Ah - excellent - thanks for confirming this issue is not endemic to this branch @jkuester !

**Comment by mrjones-plip** [comment]
@jkuester  - I think I fixed the intents with commit 2cbe625.  Could you do two builds, one as is, a second with the new line set to `<string name="app_host">local-ip.medicmobile.org</string>`?  That way we can test intents working on `local-ip.medicmobile.org` with the new build! Actually we can blindly send any matching URL to see if the intent works, even if it doesn't resolve or have a server listening there - I was able to test intent [on another instance](https://github.com/medic/cht-android/blob/echis-hack0rz/src/moh_mali_chw/res/values/strings.xml#L4) this way!

File: src/main/java/org/medicmobile/webapp/mobile/SettingsStore.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ static SettingsStore in(Context ctx) {
 		trace(SettingsStore.class, "Loading settings for context %s...", ctx);
 
 		SharedPreferences prefs = ctx.getSharedPreferences(
-				SettingsStore.class.getName(),
+				"foo",
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply src/main/java/org/medicmobile/webapp/mobile/SettingsStore.java]: Oops. Cache busting. Reverted.

**Review by jkuester — APPROVED** [review]
`#SHIP_IT`

**Review by mrjones-plip — APPROVED** [review]
With [this latest test build](https://github.com/medic/cht-android/files/11504099/echis_kenya.zip), magic links work as expected \o/. There are some caveats:
* bare domain does not work, but this should matter as we're not using it (eg `health.go.ke`)
* if you have never launched the app and have never selected an instance, magic link works and correctly takes you to the login screen
* if you have logged into instance A and get a magic link to instance B, it just takes you to instance A, still logged in
* if you use a non-existent URL, the app crashes ([see video](https://github.com/medic/cht-android/assets/8253488/8f13e1ef-8329-471a-bbb9-d7168bf00b2e))

with this i give it a very tentative "ship it"

I would very much like to get @ngaruko's input on this PR and build!

**Comment by kennsippell** [comment]
Released via https://github.com/medic/cht-android/releases/tag/untagged-9454cb4dc893a03e36f3

**Review by jkuester — APPROVED** [review]

**Pull request merged**